airflow/providers/google/cloud/hooks/bigquery.py

```diff
@@ -598,33 +597,86 @@ def create_external_table(self,  # pylint: disable=too-many-locals,too-many-argu
+            table.encryption_configuration = EncryptionConfiguration.from_api_repr(encryption_configuration)
+
+        self.log.info('Creating external table: %s', external_project_dataset_table)
+        self.create_empty_table(table_resource=table.to_api_repr(), project_id=project_id, location=location)
+        self.log.info('External table created successfully: %s', external_project_dataset_table)
+
+    @GoogleBaseHook.fallback_to_default_project_id
+    def update_table(
+        self,
+        table_resource: Dict[str, Any],
+        fields: Optional[List[str]] = None,
+        dataset_id: Optional[str] = None,
+        table_id: Optional[str] = None,
+        project_id: Optional[str] = None,
+    ) -> Dict[str, Any]:
+        """
+        Change some fields of a table.
+
+        Use ``fields`` to specify which fields to update. At least one field
+        must be provided. If a field is listed in ``fields`` and is ``None``
+        in ``table``, the field value will be deleted.
+
+        If ``table.etag`` is not ``None``, the update will only succeed if
+        the table on the server has the same ETag. Thus reading a table with
+        ``get_table``, changing its fields, and then passing it to
+        ``update_table`` will ensure that the changes will only be saved if
+        no modifications to the table occurred since the read.
 
-        service.tables().insert(  # pylint: disable=no-member
-            projectId=project_id,
-            datasetId=dataset_id,
-            body=table_resource
-        ).execute(num_retries=self.num_retries)
+        :param project_id: The project to create the table into.
+        :type project_id: str
+        :param dataset_id: The dataset to create the table into.
+        :type dataset_id: str
+        :param table_id: The Name of the table to be created.
+        :type table_id: str
+        :param table_resource: Table resource as described in documentation:
+            https://cloud.google.com/bigquery/docs/reference/rest/v2/tables#Table
+            The table has to contain ``tableReference`` or ``project_id``, ``datset_id`` and ``table_id``
+            have to be provided.
+        :type table_resource: Dict[str, Any]
+        :param fields: The fields of ``table`` to change, spelled as the Table
+            properties (e.g. "friendly_name").
+        :type fields: List[str]
+        """
+        fields = fields or list(table_resource.keys())
+        table_resource = self._resolve_table_reference(
+            table_resource=table_resource,
+            project_id=project_id,
+            dataset_id=dataset_id,
+            table_id=table_id
+        )
 
-        self.log.info('External table created successfully: %s',
-                      external_project_dataset_table)
+        table = Table.from_api_repr(table_resource)
+        self.log.info('Updating table %s.%s.%s', project_id, dataset_id, table_id)
```

Comment:
Does project_id, dataset_id, table_id always have value? These are not optional parameters? There is no alternative way to pass these values?

Reply:
Good catch, user can provide table resource